sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1331,6 +1331,16 @@ object SQLConf {
       .booleanConf
       .createWithDefault(true)
 
+  val PANDAS_ARROW_SAFE_TYPE_CONVERSION =
+    buildConf("spark.sql.execution.pandas.arrowSafeTypeConversion")
+      .internal()
+      .doc("When true, enabling Arrow do safe type conversion check when converting" +
+        "Pandas.Series to Arrow Array during serialization. Arrow will raise errors " +
+        "when detecting unsafe type conversion. When false, disabling Arrow's type " +
+        "check and do type conversions anyway.")
+      .booleanConf
+      .createWithDefault(true)
```

Comment:
If enabling by default causes a lot of the current tests to fail, it might also do the same for users - maybe we should disable by default?

Reply:
As you said, if it is false by default, it allows overflow. cc @ueshin @HyukjinKwon  What do you think? Which one is better default value?

Reply:
I'd favor true by default. Do we even need this flag? As in many such cases, I'm just not clear a) how a user would find this option and b) when they would disable it. Disabling allows something to continue that is going to also fail or give an incorrect answer, right?

Reply:
I think the big issue with this is when NULL values are introduced in an integer column. Pandas will automatically convert these to floating-points to represent the NULLs, then when Arrow casts it back to integer, it will raise an error due to truncation - I don't think Arrow checks the actual values, but maybe it should?  For example, with safe=True:

```python
>>> pa.Array.from_pandas(pd.Series([1, None]), type=pa.int32(), safe=True)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "pyarrow/array.pxi", line 474, in pyarrow.lib.Array.from_pandas
  File "pyarrow/array.pxi", line 169, in pyarrow.lib.array
  File "pyarrow/array.pxi", line 69, in pyarrow.lib._ndarray_to_array
  File "pyarrow/error.pxi", line 81, in pyarrow.lib.check_status
pyarrow.lib.ArrowInvalid: Floating point value truncated
```

Reply:
Does Spark check for these types of errors with standard udfs, when not using Arrow?

Reply:
I think for standard udfs, there is no similar type check:

```python
>>> test_udf = udf(lambda x: 128, "byte")
>>> spark.range(0, 1).select(test_udf('id')).show()
+------------+
|<lambda>(id)|
+------------+
|        -128|
+------------+
```

Reply:
For arrow users, the old behavior is to have some errors when overflows happen. I think this is what users expect. If we make this config false by default, we change this behavior. This sounds not pretty safe.

Although we change behavior too if we make the config true by default. The behavior is changed by raising errors in cases including NULL values in an integer column. Because we show some messages and users can disable this config if they want.

We will change behavior either using true or false as default config. Sounds like making it as true is a bit safer.

Reply:
https://issues.apache.org/jira/browse/ARROW-4258

Reply:
Thanks @kszucs

Reply:
No .. I was referring https://github.com/apache/spark/pull/22807#discussion_r247487281 ..

Reply:
> So, NULL -> Integer issue will be fixed in Arrow 0.12.0?

Yes, I verified this and ran the unit tests on the release